log4j-core/src/main/java/org/apache/logging/log4j/core/config/Property.java

```diff
@@ -118,15 +119,38 @@ public static Property createProperty(final String name, final String value) {
      * @param value The value.
      * @return A Property.
      */
+    public static Property createProperty(
+            final String name,
+            final String rawValue,
+            final String value) {
+        if (name == null) {
+            LOGGER.error("Property name cannot be null");
```

Comment:
If it's an error and we continue as if nothing happened, then either this should not be logged as an ERROR or some error-like behavior should happen. It's just weird to log an ERROR and then continue as if nothing happened.

Reply:
Agree that this is odd, it matches the behavior of the previous plugin-based constructor. I have to imagine anything that receives a Property instance with a null key will throw, so it's probably best to throw in this case. I can update all the constructors